src/ir/iteration.h

```diff
@@ -40,8 +40,21 @@ namespace wasm {
 //                        this instruction. For example, includes If::condition
 //                        but not If::ifTrue.
 //
-template<class Specific> class AbstractChildIterator {
-  using Self = AbstractChildIterator<Specific>;
+// ChildPointerIterator and ValueChildPointerIterator iterate the same way
+// respectively, but they return pointers to children instead.
+
+template<bool isPointer> struct DereferenceKind {};
+template<> struct DereferenceKind<true> {
```

Comment:
We could slightly simplify this by not using a `bool` and instead passing these structs directly as the template argument that determines what kind of dereference is done. Sorry for not thinking of this yesterday!

```suggestion
struct PointerDereference {
```